chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -99,9 +97,23 @@ impl Ledger {
                 // update ledger state
                 self.accounts = new_state.clone();
                 self.logs = new_logs.clone();
-                Ok(())
             }
         }
+
+        self.update_environment();
+        Ok(())
+    }
+    fn update_environment(&mut self) {
+        let next_number = self.environment.block_number + BlockNumber::one();
+        // this is a simplified block hash calculation
+        let next_hash: BlockHash = <[u8; 32]>::from(next_number).into();
```

Comment:
I assume we can get `next_hash` from the real block hash value

Reply:
Good point. I'm taking a look at how this can be done.

Reply:
I changed the signature for this function, it now uses the `metadata.content_hash` value, converted into a `BlockHash` type, inserts it into the `block_hashes` Vec, and updates the `block_number`.